### ISSUE TITLE
This library is buggy.

### ISSUE THREAD

**Issue by dpc** [grafov/bcast#12]
https://github.com/grafov/bcast/blob/master/bcast.go#L114

```
go func(out chan interface{}, received *Message) { // non blocking
     out <- received.payload
}(member, &received)
```

This code can block, and keep the goroutine around, effectively being a memory leak.

Along with other gotchas (like reordering the messages), and other race condition, it makes this library unsuitable for any serious usage.

While I appreciate your effort and know this software comes with no warranty, please indicate that this library is buggy in README, delete it or accept the PRs from various sources addressing problems with it.

**Comment by grafov** [comment]
Thank for you note, I left the library as is but it seems it need more time for refactoring.

**Comment by grafov** [comment]
It seems it was fixed by PR #11. So I close the issue. Left the comment if the bug still exists. Thank you for reporting.

**Comment by dpc** [comment]
Please don't mind me originally complaining about it. It's great that it got fix, and I hope you will keep maintaining it!
